app/assets/javascripts/host_edit.js

```diff
@@ -75,24 +75,41 @@ function computeResourceSelected(item){
 }
 
 function update_capabilities(capabilities){
+  capabilities = capabilities.split(" ")
   $('#image_provisioning').empty();
   $('#image_selection').appendTo($('#image_provisioning'));
   update_provisioning_image();
   $('#manage_network').empty();
   $('#subnet_selection').appendTo($('#manage_network'));
 
-  var build = (/build/i.test(capabilities));
-  var image = (/image/i.test(capabilities));
-  if (build){
+  $('input[id^=host_provision_method_]').attr('disabled', true);
+  for (i = 0; i < capabilities.length; i++) {
+    $('input[id^=host_provision_method_' + capabilities[i] + ']').attr('disabled', false);
+  }
+
+  var build = (capabilities.indexOf('build') > -1);
+  var image = (capabilities.indexOf('image') > -1);
+  if (build) {
     $('#manage_network_build').show();
     $('#host_provision_method_build').click();
     build_provision_method_selected();
-  } else {
+  } else if(image) {
     $('#manage_network_build').hide();
```

Comment:
We could delete this, as the next else case does the same. But that would make it harder to read. Any suggestions what to do?